sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetScan.scala

```diff
@@ -2517,11 +2536,25 @@ class MultiFileCloudParquetPartitionReader(
     }
     val colTypes = readDataSchema.fields.map(f => f.dataType)
 
-    RmmRapidsRetryIterator.withRetryNoSplit(hostBuffer) { _ =>
+    var currentTargetBatchSize = targetBatchSizeBytes
+    val splitBatchSizePolicy: HostMemoryBuffer => Seq[HostMemoryBuffer] = hostBuffer => {
+      if (!useChunkedReader) {
+        throw new SplitAndRetryOOM("GPU OutOfMemory: could not split inputs " +
+            "chunked parquet reader is configured off")
+      }
+      currentTargetBatchSize = currentTargetBatchSize / 2
+      if (currentTargetBatchSize < 10 * 1024 * 1024) { // 10 MiB
+        throw new SplitAndRetryOOM("GPU OutOfMemory: could not split input " +
+            "target batch size to less than 10 MiB")
+      }
+      Seq(hostBuffer)
+    }
+
+    RmmRapidsRetryIterator.withRetry(hostBuffer, splitBatchSizePolicy) { _ =>
```

Comment:
I think we should fix the API to support a case like this, where we don't want to return a Seq[T]. Could be done separately, as it's probably easy enough to search for `withRetry` and take care of it in one go.

Reply:
I am not sure what other cases we would want this.  If you can think of one, then we can do it.  Otherwise I would rather wait until we run into that case.

Reply:
That sounds good to me